platform/backend/repository/sqlite/src/main/kotlin/io/hamal/repository/sqlite/record/func/Repository.kt

```diff
@@ -80,6 +80,30 @@ class SqliteFuncRepository(
         }
     }
 
+    override fun deploy(funcId: FuncId, cmd: FuncCmdRepository.DeployCmd): Func {
+        val cmdId = cmd.id
+        return tx {
+            if (commandAlreadyApplied(cmdId, funcId)) {
+                versionOf(funcId, cmdId)
+            } else {
+                val currentVersion = versionOf(funcId, cmdId)
+                if (cmd.versionToDeploy !in CodeVersion(1)..currentVersion.code.version) {
+                    throw NoSuchElementException("${cmd.versionToDeploy} does not exist")
+                }
```

Comment:
CodeVersion is an IntValueObject ---> implements comparable - make sure the behavior for memory version is the same

```suggestion
require(cmd.versionToDeploy <= currentVersion.code.version) {"${cmd.versionToDeploy} can not be deployed"}
```

Reply:
Good call to have the validation here inside of the transaction - better than my suggestion :+1: